distribution/metadata/v2_metadata_service.go

```diff
@@ -15,10 +18,24 @@ type V2MetadataService struct {
 
 // V2Metadata contains the digest and source repository information for a layer.
 type V2Metadata struct {
+	AuthConfigHash   string
 	Digest           digest.Digest
 	SourceRepository string
 }
 
+// HashAuthConfig returns an sha256 sum of the "authConfig" if successful.
+func HashAuthConfig(authConfig *types.AuthConfig) (string, error) {
+	if authConfig == nil {
+		return "", nil
+	}
+	buf, err := json.Marshal(authConfig)
```

Comment:
Unfortunately, using this wouldn't help OpenShift. We use dummy user name almost everywhere. We differentiate based on the password which is a bearer token passed to OpenShift API which determines user's identity.

We could address it on our side though if you are strongly against storing the password.